crates/l2/contracts/src/l1/OnChainProposer.sol

```diff
@@ -92,11 +93,11 @@ contract OnChainProposer is IOnChainProposer, ReentrancyGuard {
             "OnChainProposer: block not committed"
         );
         require(
-            !verifiedBlocks[blockNumber],
+            blockNumber > lastVerifiedBlock,
```

Comment:
I think `blockNumber` should be equal to `lastVerifiedBlock + 1`. Otherwise, someone could verify a later block and then middle blocks will not be verifiable anymore.

Reply:
ah yeah. Applied the fix!